components/motor/fake/motor.go

```diff
@@ -251,25 +251,25 @@ func (m *Motor) Direction() int {
 	return 0
 }
 
-// If revolutions is 0, the returned wait duration will be 0 representing that
-// the motor should run indefinitely.
 func goForMath(maxRPM, rpm, revolutions float64) (float64, time.Duration, float64) {
 	// need to do this so time is reasonable
 	if rpm > maxRPM {
 		rpm = maxRPM
 	} else if rpm < -1*maxRPM {
 		rpm = -1 * maxRPM
 	}
```

Comment:
so since having `rpm = 0` or `revolutions = 0` are now invalid requests, maybe we can do something
```
func goForMath(maxRPM, rpm, revolutions float64) (float64, time.Duration, float64) {
        dir := 1.0
        
	// check that a valid request has been made
	if rpm == 0 || revolutions == 0{
	return 0, 0, dir
	}

        // need to do this so time is reasonable
	if rpm > maxRPM {
		rpm = maxRPM
	} else if rpm < -1*maxRPM {
		rpm = -1 * maxRPM
	}
	
	// flip sign if needed
	if rpm*revolutions < 0 {
	     dir = -1.0
	}
	
	powerPct := math.Abs(rpm) / maxRPM * dir
	waitDur := time.Duration(math.Abs(revolutions/rpm)*60*1000) * time.Millisecond
	return powerPct, waitDur, dir
}
```

Reply:
we can also just return during the dir = 0 case instead, to have the same behavior

Reply:
our pattern before this was to do those checks in GoFor, so I added a revolutions check to the exiting sped check but it's just not in goForMath